WordPress/Classes/Stores/PluginStore+Persistence.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+
+extension PluginStoreState: Codable {
+
+    private enum CodingKeys: String, CodingKey {
+        case plugins
+        case featuredPluginSlugs
+        case directoryFeeds
+        case directoryEntries
+    }
+
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        plugins = try container.decode([JetpackSiteRef: SitePlugins].self, forKey: .plugins)
+        featuredPluginsSlugs = try container.decode([String].self, forKey: .featuredPluginSlugs)
+        directoryFeeds = try container.decode([String: PluginDirectoryPageMetadata].self, forKey: .directoryFeeds)
+        directoryEntries = try container.decode([String: PluginDirectoryEntryState].self, forKey: .directoryEntries)
+    }
+
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        try container.encode(plugins, forKey: .plugins)
+        try container.encode(featuredPluginsSlugs, forKey: .featuredPluginSlugs)
+        try container.encode(directoryFeeds, forKey: .directoryFeeds)
+        try container.encode(directoryEntries, forKey: .directoryEntries)
+    }
+
+}
+
+extension PluginStore {
+
+    private static let cacheFilename = "plugins.json"
+
+    func writeCachedJSON() {
```

Comment:
I see this is only called from `queriesChanged`, which should be enough in most cases but might mean having stale cache data if the app is killed (crash or force quit). Have you considered that scenario? (I'm not really sure it'd be a big deal, but worth thinking about)